crypto/fipsmodule/ml_kem/mlkem/mlkem_native.h

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024-2025 The mlkem-native project authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MLK_H
+#define MLK_H
+/*
+ * Public API for mlkem-native
+ *
+ * This header defines the public API of a single build of mlkem-native.
+ *
+ * To use this header, make sure one of the following holds:
+ *
+ * - The config.h used for the build is available in the include paths.
+ * - The values of MLK_BUILD_INFO_LVL and MLK_BUILD_INFO_NAMESPACE are set,
+ * reflecting the security level (512/768/1024) and namespace of the build.
+ *
+ * This header specifies a build of mlkem-native for a fixed security level.
+ * If you need multiple builds, e.g. to build a library offering multiple
+ * security levels, you need multiple instances of this header. In this case,
+ * make sure to rename or #undefine the header guard
+ */
+
+#include <stdint.h>
+
+/*************************** Build information ********************************/
+
+/*
+ * Provide security level (MLK_BUILD_INFO_LVL) and namespacing
+ * (MLK_BUILD_INFO_NAMESPACE)
+ *
+ * By default, this is extracted from the configuration used for the build,
+ * but you can also set it manually to avoid a dependency on the build config.
+ */
+
+/* Skip this if MLK_BUILD_INFO_LVL has already been set */
+#if !defined(MLK_BUILD_INFO_LVL)
+
+/* Option 1: Extract from config */
+#if defined(MLK_CONFIG_FILE)
+#include MLK_CONFIG_FILE
+#else
+#include "config.h"
+#endif
+
+#if MLKEM_K == 2
+#define MLK_BUILD_INFO_LVL 512
+#elif MLKEM_K == 3
+#define MLK_BUILD_INFO_LVL 768
+#elif MLKEM_K == 4
+#define MLK_BUILD_INFO_LVL 1024
+#else
+#error MLKEM_K not set by config file
+#endif
+
+#ifndef MLK_NAMESPACE_PREFIX
+#error MLK_NAMESPACE_PREFIX not set by config file
+#endif
+
+#if defined(MLK_MULTILEVEL_BUILD_WITH_SHARED) || \
+    defined(MLK_MULTILEVEL_BUILD_NO_SHARED)
+#define MLK_BUILD_INFO_CONCAT3_(x, y, z) x##y##_##z
```

Comment:
The `_` before `y` is left out intentionally, I assume.

Reply:
Yes, the level (512/768/1024) is directly appended to the namespace prefix.